### PR TITLE
Hocs 1125 create api routes and middleware

### PR DESCRIPTION
**Pull request by Raysonviswas**

### CODE REVIEW

File: server/lists/adapters/units.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const byDisplayName = (a, b) => a.displayName.toUpperCase().localeCompare(b.displayName.toUpperCase());
+
+module.exports = async (data, { logger }) => {
+    logger.debug('REQUEST_UNITS', { units: data.length });
+    logger.debug('data', data);
```
Comment (lewismuirhead) [inline server/lists/adapters/units.js]: remove this

File: src/shared/router/routes/index.ts [hunk 2]
```diff
@@ -47,7 +48,12 @@ const routes = [
         path: '/add-child-topic',
         exact: true,
         component: AddChildTopic,
-        title: 'Add Unit'
+        title: 'Add Unit'},
```
Comment (lewismuirhead) [inline src/shared/router/routes/index.ts:73]: Spacing not quite right here - looks like a merge issue

File: src/shared/services/unitsService.ts [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+import Item from 'shared/models/item';
 
 export const createUnit = (unit: Unit) => new Promise((resolve, reject) => axios
     .post('/api/units', unit)
     .then(() => resolve())
     .catch(reason => reject(reason))
 );
+
+export const getUnits = () => new Promise<[Item]>((resolve, reject) =>
+    axios.get('/api/units')
```
Comment (lewismuirhead) [inline src/shared/services/unitsService.ts]: looks good but needs some unit tests

**Review by lewismuirhead — CHANGES_REQUESTED** [review]

File: src/shared/pages/unit/unitSearch/unitSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        })), []);
+
+    return (
+        <div className="govuk-form-group">
+            <Link className="govuk-back-link" to="/">Back</Link>
+            <ErrorSummary pageError={pageError} />
+            <h1 className="govuk-heading-xl">
+                Unit search
+            </h1>
+            <TypeAhead
```
[the remark this answers is not in the document]
Comment (Raysonviswas) [reply src/shared/pages/unit/unitSearch/unitSearch.tsx:56]: yes

**Pull request closed without merging**